markdown/ch7.markdown

```diff
@@ -342,12 +342,13 @@ def align(reference, target, cost=cost_mse):
         res = optimize.minimize(cost, p, args=(ref, tgt), method='Powell')
         p = res.x
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print('Level: %i, Angle: %.2f, Offset: (%.1f, %.1f), Cost: %.2f' %
```

Comment:
Yes, formatting is terrible, I don't know who thought it was an improvement. f-strings, on the other hand, might be nice, but then we are 3.6-only:

```python
print(f'Level: {n}, Angle: {np.rad2deg(res.x[0]):.3}, '
      f'Offset: ({res.x[1] * 2**n:.3}, {res.x[2] * 2**n:.3}), '
      f'Cost: {res.fun:.3}', end='\r')
```